### ISSUE TITLE
Mouse Cursor Support

### ISSUE THREAD

**Issue by philvessey** [arnaudleclerc/AzureMapsControl.Components#37]
Hi

I have a bubble layer setup which is clustering symbols and working well.
I have setup the LayerEventType.MouseEnter and LayerEventType.MouseLeave on the layer and have the following code for the events:

bubbleLayer.OnMouseEnter += e =>
{
    //Change mouse cursor to pointer
};

bubbleLayer.OnMouseLeave += e =>
{
    //Change mouse cursor to grab
};

Is there any way to manipulate the mouse cursor from these events?

**Comment by philvessey** [comment]
The JS to achieve the same would be:

map.events.add('mouseenter', bubbleLayer, function () {
    map.getCanvasContainer().style.cursor = 'pointer';
});

map.events.add('mouseleave', bubbleLayer, function () {
    map.getCanvasContainer().style.cursor = 'grab';
});

Thats the effect Im looking to achieve if its possible 👍

**Comment by arnaudleclerc** [comment]
Hi @philvessey, this is currently not possible. What I can imagine in the future is to add a `SetMapCanvasPropertiesAsync` API on the `Map` accepting a dictionary where the key is the name of the property you want to update. I'll mark this issue as enhancement, it can then be taken care of when the v1 has been released. Would that be ok for you?

**Comment by philvessey** [comment]
Yep sounds good.

**Comment by arnaudleclerc** [comment]
@philvessey I just published a v1.2.0 to include some methods on the `Map` class : 

- `SetCanvasStylePropertyAsync` and `SetCanvasStylePropertiesAsync` to add properties on the canvas of the map.
- `SetCanvasContainerStylePropertyAsync` and `SetCanvasContainerStylePropertiesAsync` to add properties on the container of the canvas of the map.

The main difference being that the first two methods will apply the properties on `map.getCanvas().style` where the second set of methods will apply them on `map.getCanvasContainer().style`.

Here is an example which covers your use case : 

```
@page "/BubbleLayerOnReady"

@using AzureMapsControl.Components.Map
<AzureMap Id="map"
          CameraOptions="new CameraOptions { Zoom = 2 }"
          EventActivationFlags="MapEventActivationFlags
                                .None()
                                .Enable(MapEventType.Ready)"
          OnReady="OnMapReady" />

@code  {
    public async Task OnMapReady(MapEventArgs eventArgs)
    {
        const string dataSourceId = "bubbleDataSource";
        var dataSource = new AzureMapsControl.Components.Data.DataSource(dataSourceId);
        await eventArgs.Map.AddSourceAsync(dataSource);


        var geometries = new List<AzureMapsControl.Components.Atlas.Shape>();
        for (var i = 0; i < 10; i++)
        {
            geometries.Add(new AzureMapsControl.Components.Atlas.Shape<AzureMapsControl.Components.Atlas.Point>(new AzureMapsControl.Components.Atlas.Point(new Components.Atlas.Position(i * 5, i * 5))));
        }

        await dataSource.AddAsync(geometries);

        var layer = new AzureMapsControl.Components.Layers.BubbleLayer
        {
            Options = new Components.Layers.BubbleLayerOptions
            {
                Color = new Components.Atlas.ExpressionOrString("white"),
                Radius = new Components.Atlas.ExpressionOrNumber(5),
                StrokeColor = new Components.Atlas.ExpressionOrString("#4288f7"),
                StrokeWidth = new Components.Atlas.ExpressionOrNumber(6),
                Source = dataSourceId
            },
            EventActivationFlags = Components.Layers.LayerEventActivationFlags.None().Enable(Components.Layers.LayerEventType.MouseEnter, Components.Layers.LayerEventType.MouseLeave)
        };

        layer.OnMouseEnter += async _ => {
            await eventArgs.Map.SetCanvasContainerStylePropertiesAsync(new Dictionary<string, string> {
                { "cursor", "pointer" }
            });
        };

        layer.OnMouseLeave += async _ => {
            await eventArgs.Map.SetCanvasContainerStylePropertiesAsync(new Dictionary<string, string> {
                { "cursor", "grab" }
            });
        };

        await eventArgs.Map.AddLayerAsync(layer);
    }
}
```

I'll close this issue. As usual, feel free to reopen it if necessary!

**Comment by philvessey** [comment]
Nice - will try it later on!
